### PR TITLE
[MIG][14.0]point_of_sale: migration script

### PR DESCRIPTION
**Pull request by hieu1211**
This PR
======
Done

### CODE REVIEW

File: openupgrade_scripts/scripts/point_of_sale/14.0.1.0.1/upgrade_analysis_work.txt [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+ir.rule: point_of_sale.rule_pos_bank_statement_line_user (noupdate) (noupdate switched)
+ir.rule: point_of_sale.rule_pos_bank_statement_user (noupdate) (noupdate switched)
+ir.rule: point_of_sale.rule_pos_cashbox_line_accountant (noupdate) (noupdate switched)
+ir.rule: point_of_sale.rule_pos_config_multi_company (noupdate) (noupdate switched)
+ir.rule: point_of_sale.rule_pos_multi_company (noupdate) (noupdate switched)
+ir.rule: point_of_sale.rule_pos_order_report_multi_company (noupdate) (noupdate switched)
+ir.rule: point_of_sale.rule_pos_payment_method_multi_company (noupdate) (noupdate switched)
+ir.rule: point_of_sale.rule_pos_payment_multi_company (noupdate) (noupdate switched)
+ir.rule: point_of_sale.rule_pos_session_multi_company (noupdate) (noupdate switched)
+# DONE: pre-migration change noupdate
```
Comment (MiquelRForgeFlow) [inline openupgrade_scripts/scripts/point_of_sale/14.0.1.0.1/upgrade_analysis_work.txt:89]: where is it done?

File: openupgrade_scripts/scripts/point_of_sale/14.0.1.0.1/upgrade_analysis_work.txt [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+point_of_sale / res.partner              / barcode (char)                : module is now 'base' ('point_of_sale')
+point_of_sale / res.partner              / barcode (char)                : not stored anymore
+# NOTHING TO DO
+
+point_of_sale / stock.picking            / pos_order_id (many2one)       : NEW relation: pos.order
+# DONE: post-migration convert data many2one to one2many
+
+point_of_sale / stock.picking            / pos_session_id (many2one)     : NEW relation: pos.session
+# NOTHING TO DO: new feature
```
Comment (MiquelRForgeFlow) [inline openupgrade_scripts/scripts/point_of_sale/14.0.1.0.1/upgrade_analysis_work.txt]:
```suggestion
# DONE: post-migration: ....
```
as you have done a script for this

File: openupgrade_scripts/scripts/point_of_sale/14.0.1.0.1/upgrade_analysis_work.txt [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+NEW ir.model.access: point_of_sale.access_pos_details_wizard
+NEW ir.model.access: point_of_sale.access_pos_make_payment
+NEW ir.model.access: point_of_sale.access_product_attribute_custom_value_pos_manager
+NEW ir.model.access: point_of_sale.access_product_attribute_pos_manager
+NEW ir.model.access: point_of_sale.access_product_attribute_value_pos_manager
+NEW ir.model.access: point_of_sale.access_product_product_attribute_pos_manager
+NEW ir.model.access: point_of_sale.access_product_template_attribute_exclusion_pos_manager
+NEW ir.model.access: point_of_sale.access_product_template_attribute_line_pos_manager
+DEL ir.model.access: pos_cash_rounding.access_account_cash_rounding
+DEL ir.model.constraint: point_of_sale.constraint_res_partner_unique_barcode
```
Comment (MiquelRForgeFlow) [inline openupgrade_scripts/scripts/point_of_sale/14.0.1.0.1/upgrade_analysis_work.txt]: you have to delete it. see https://github.com/OCA/OpenUpgrade/commit/8756d0413ee2794bdb737300a1de6bf9b46453c5 as example

**Comment by hieu1211** [comment]
@MiquelRForgeFlow Done, thanks for reviewing

**Review by MiquelRForgeFlow — APPROVED** [review]

File: openupgrade_scripts/scripts/point_of_sale/14.0.1.0.1/upgrade_analysis_work.txt [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+point_of_sale / pos.payment              / cardholder_name (char)        : NEW
+point_of_sale / pos.payment              / is_change (boolean)           : NEW hasdefault
+point_of_sale / pos.payment              / payment_status (char)         : NEW
+point_of_sale / pos.payment              / ticket (char)                 : NEW
+point_of_sale / pos.payment.method       / active (boolean)              : NEW hasdefault
+point_of_sale / pos.session              / cash_real_difference (float)  : NEW
+point_of_sale / pos.session              / cash_real_expected (float)    : NEW
+point_of_sale / pos.session              / cash_real_transaction (float) : NEW
+point_of_sale / pos.session              / picking_ids (one2many)        : NEW relation: stock.picking
+point_of_sale / pos.session              / state (selection)             : selection_keys is now '['closed', 'closing_control', 'opened', 'opening_control']' ('['closed', 'closing_control', 'new_session', 'opened', 'opening_control']')
```
Comment (royle-vietnam) [inline openupgrade_scripts/scripts/point_of_sale/14.0.1.0.1/upgrade_analysis_work.txt:37]: @hieu1211 i think need to map 'new_session' to 'opening_control' / 'opened'

Comment (MiquelRForgeFlow) [reply openupgrade_scripts/scripts/point_of_sale/14.0.1.0.1/upgrade_analysis_work.txt:37]: Yeah, new_session has to be mapped to one of the other options.

Comment (hieu1211) [reply openupgrade_scripts/scripts/point_of_sale/14.0.1.0.1/upgrade_analysis_work.txt:37]: i missed it, thanks!

**Comment by cvinh** [comment]
Hello
I don't see the script to fill pos.payment object and pos.order.line.full_product_name
Best regards

**Comment by legalsylvain** [comment]
hi @hieu1211. Thanks for sharing this work. 
Did you took the time to take a look on the @cvinh remarks ? 

thanks !

**Comment by hieu1211** [comment]
> hi @hieu1211. Thanks for sharing this work. Did you took the time to take a look on the @cvinh remarks ?
> 
> thanks !

Thanks for the reminder, i will check and fix soon!

**Comment by MiquelRForgeFlow** [comment]
Poor bot. Nobody gives it access.

**Comment by MiquelRForgeFlow** [comment]
Superseded by https://github.com/OCA/OpenUpgrade/pull/3499.

**Pull request closed without merging**